File: C++/Remove Half Nodes.cpp
class Solution {
  public:
    Node *RemoveHalfNodes(Node *root) {
        if(!root) return NULL;
        root->left = RemoveHalfNodes(root->left);
        root->right = RemoveHalfNodes(root->right);
        Node *temp = root;
        if((!root->left and root->right) or (!root->right and root->left)){
            if(root->left) temp = root->left;
            else temp = root->right;
            root = NULL;
        }
        return temp;
    }
};

File: JAVA/Remove Half Nodes.java
class Solution {
    public Node help(Node node){
      if(node==null)return null;
      node.left=help(node.left);
      node.right=help(node.right);
      if(node.left==null && node.right==null) return node;
      else if(node.right==null) return node=node.left;
      else if(node.left==null)  return  node=node.right; 
    return node;
    }
    public Node RemoveHalfNodes(Node root) {
        Node ans=help(root);
        return ans;
    }
  }
